### PR TITLE
refactor browser: remove browser pane - browser dependency completely

### PR DESCRIPTION
**Pull request by TAKeanice**
by moving WebUIDelegate implementation to browser, where it belongs

also:
- browser is not a view but a controller, make NSObject subclass
- rename browserView such that it actually is called browser (which it is)

### CODE REVIEW

**Comment by josh64x2** [comment]
@TAKeanice we'll put this one in the 3.6.0 milestone since 3.5.0 is almost complete. Love the changes though 👍

File: Vienna/Sources/Main window/BrowserPane.m [hunk 1]
```diff
@@ -779,4 +518,179 @@ -(BOOL)hasCloseButton
...
+
+#pragma mark - WebUIDelegate helper methods
+
+- (void)hoveredOverURL:(NSURL *)url {
+    if (self.statusBar) {
+        self.statusBar.label = url.absoluteString;
+    }
+}
+
+- (void)setIcon:(NSImage *)icon {
```
Comment (barijaona) [inline Vienna/Sources/Main window/BrowserPane.m]: Is it really a WebUIDelegate helper method ?
It's only called once, and from this class itself.

**Comment by TAKeanice** [comment]
I think it is totally fine to publish it in a later release, there should only not be changes to the corresponding code that gives conflicts at a later merge.

File: Vienna/Sources/Main window/BrowserPane.m (hunk 1 above)
Comment (TAKeanice) [reply Vienna/Sources/Main window/BrowserPane.m]: You are right, it seems not to be a delegate method, just slipped into the wrong section. I probably confused it with `receivedIcon` from the `WebFrameLoadDelegte`

Comment (TAKeanice) [reply Vienna/Sources/Main window/BrowserPane.m]: These methods are in the `WebFrameLoadHelperMethods` section, not in the `WebUIDelegate` section

Comment (TAKeanice) [reply Vienna/Sources/Main window/BrowserPane.m]: Probably I should inline it.

**Comment by TAKeanice** [comment]
In preparation of implementing a new browser backed by the WebKit framework, I cleaned up the interface of the current browser. More changes will come, mostly to reduce the amount of calls to `primaryTab` and `activeTab` from outside the browser. Most things for which these attributes are used should happen in the browser itself, and not be initiated from another object, since they rely heavily on assumptions about the browser.

**Comment by TAKeanice** [comment]
Work on this has proceeded and became a full browser refactoring with a Webkit view replacing the Webview. Anyhow, as a first step for a cleaner and ultimatively new browser, this is ready to be merged

**Pull request merged**